pkg/scm/bitbucket.go

```diff
@@ -0,0 +1,249 @@
+package scm
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/ktrysmt/go-bitbucket"
+	"github.com/pluralsh/oauth"
+	"github.com/pluralsh/plural/pkg/utils"
+)
+
+var (
+	BitbucketClientSecret string
+)
+
+const emailError = "Can't find the user email address"
+
+type Link struct {
+	Name string `json:"name"`
+	Href string `json:"href"`
+}
+
+type Values struct {
+	Email     string `json:"email"`
+	IsPrimary bool   `json:"is_primary"`
+}
+
+type Bitbucket struct {
+	Client *bitbucket.Client
+}
+
+func (b *Bitbucket) Init() error {
+	flow := &oauth.Flow{
+		Host: &oauth.Host{
+			AuthorizeURL: "https://bitbucket.org/site/oauth2/authorize",
+			TokenURL:     "https://bitbucket.org/site/oauth2/access_token",
+		},
+		ClientID:     "GVEHgz5FMA24BdcKfA",
+		ClientSecret: BitbucketClientSecret,
+		CallbackURI:  "http://127.0.0.1/callback",
+		ResponseType: "code",
+	}
+
+	accessToken, err := flow.WebAppFlow()
+	if err != nil {
+		return err
+	}
+	b.Client = bitbucket.NewOAuthbearerToken(accessToken.Token)
+
+	return nil
+}
+
+func (b *Bitbucket) Setup() (Context, error) {
+	user, err := b.Client.User.Profile()
+	if err != nil {
+		return Context{}, err
+	}
+	emails, err := b.Client.User.Emails()
+	if err != nil {
+		return Context{}, err
+	}
+	emailAddress := ""
+	emailValues, ok := emails.(map[string]interface{})
+	if !ok {
+		return Context{}, fmt.Errorf(emailError)
+	}
+	emailAddress, err = getEmailAddress(emailValues)
+	if err != nil {
+		return Context{}, err
+	}
+
+	wl, err := b.Client.Workspaces.List()
+	if err != nil {
+		return Context{}, err
+	}
+
+	workspaces := make([]string, 0)
+	for _, w := range wl.Workspaces {
+		workspaces = append(workspaces, w.Slug)
+	}
+
+	if len(workspaces) == 0 {
+		return Context{}, fmt.Errorf("You don't have any Bitbucket workspace created. Please create one first \n")
+	}
+
+	workspace := workspaces[0]
+	if len(workspaces) > 1 {
+		prompt := &survey.Select{
+			Message: "Select the workspace:",
+			Options: workspaces,
+		}
+		if err := survey.AskOne(prompt, &workspace, survey.WithValidator(survey.Required)); err != nil {
+			return Context{}, err
+		}
+	}
+
+	pr, err := b.Client.Workspaces.Projects(workspace)
+	if err != nil {
+		return Context{}, err
+	}
+	projectKeys := make(map[string]string, 0)
+	projects := make([]string, 0)
+	for _, p := range pr.Items {
+		projectKeys[p.Name] = p.Key
+		projects = append(projects, p.Name)
+	}
+
+	project := ""
+	if len(projects) == 0 {
+		return Context{}, fmt.Errorf("You don't have any Bitbucket project created. Please create one first \n")
```

Comment:
reword to `You don't have a BitBucket project created, please create one and try again` perhaps